src/origin_sdk/service/Scenario.py

```diff
@@ -277,15 +279,17 @@ def get_latest_scenario_from_region(
                 )
 
             else:
-                # Lowercase the name filter if it exists
-                name_filter = name_filter.lower()
+                # Handle the singular name_filter into a multiple
+                name_filter = [name_filter] if type(name_filter) == str else name_filter
 
                 # Now filter the list of scenarios by the name filter if it exists
                 latest = [
                     scenario
                     for scenario in regional_scenarios
-                    if name_filter is None
-                    or name_filter in scenario.get("name").lower()
+                    if all(
+                        filter.lower() in scenario.get("name").lower()
+                        for filter in name_filter
+                    )
                 ][0]
```

Comment:
Since we only need to find one scenario here, wouldn't it be ideal to use a generator expression like
```latest = next((...scenario for ...), None)```

This way we don't create a intermediate list just to stop at the first matching element. I guess it also depends on how big regional_scenarios is.

Reply:
👍